pytorch_lightning/utilities/auto_restart.py

```diff
@@ -741,6 +749,97 @@ def _teardown_dataloader_get_iterators() -> None:
         del DataLoader._ori_get_iterator
 
 
+def _validate_iterable_dataset(dataloader: DataLoader) -> Tuple[bool, str]:
+
+    SUPPORTED_SAMPLERS = (RandomSampler, SequentialSampler, DistributedSampler)
+
+    dataset = dataloader.dataset
+
+    if getattr(dataset, "__next__", None) is None:
+        raise AttributeError(
+            "Fault Tolerant Training doesn't support an IterableDataset without a `__next__` "
+            "method implemented. Hint: We recommend you to move your logic inside and rely "
+            "on a sampler, generator to perform the iteration."
```

Comment:
We should stop using "training" as it applies to non-training too

```suggestion
            "Fault-tolerance doesn't support an `IterableDataset` without `__next__` "
            "implemented. Hint: We recommend you to move your logic inside and rely on"
            "on a sampler, generator to perform the iteration."
```

Also, the hint could be clarified further

Reply:
Done !